be/src/vec/spill/spill_writer.cpp

```diff
@@ -76,7 +71,7 @@ Status SpillWriter::write(const Block& block, size_t& written_bytes) {
         auto tmp_block = block.clone_empty();
         const auto& src_data = block.get_columns_with_type_and_name();
 
-        for (size_t row_idx = 0; row_idx < rows;) {
+        for (size_t row_idx = 0; row_idx < rows && !state->is_cancelled();) {
```

Comment:
There is some atomic variable in state is cancelled method and maybe some lock.
I think you do not need check it every row, maybe every 4096 rows.

Reply:
it's only checked every batch_size rows